internal/components/components.go

```diff
@@ -213,6 +214,7 @@ func Components() (component.Factories, error) {
 		datadogexporter.NewFactory(),
 		dynatraceexporter.NewFactory(),
 		elasticexporter.NewFactory(),
+		elasticsearchexporter.NewFactory(),
```

Comment:
This might not be effective anymore for 0.42.0. If you want the ES exporter to be part of the contrib distribution, make sure it's included in this file:

https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml